src/main/js/main_html5.js

```diff
@@ -1131,24 +1154,71 @@ require("../../../html5-common/js/utils/environment.js");
      * @method OoyalaVideoWrapper#checkForClosedCaptions
      */
     var checkForClosedCaptions = _.bind(function() {
-      if (_video.textTracks && _video.textTracks.length > 0) {
-        var languages = [];
-        for (var i = 0; i < _video.textTracks.length; i++) {
-          if (((OO.isSafari || OO.isEdge) && isLive) || _video.textTracks[i].kind === "captions") {
-            var captionInfo = {
-              language: "CC",
-              inStream: true,
-              label: "In-Stream"
-            };
-            //Don't overwrite other closed captions of this language. They have priority.
-            if (availableClosedCaptions[captionInfo.language] == null) {
-              addClosedCaptions(captionInfo);
-            }
+      if (!_video.textTracks || !_video.textTracks.length) {
+        return;
+      }
+      var expectingStreamTextTracks = (OO.isSafari || OO.isEdge) && isLive;
+
+      Array.prototype.forEach.call(_video.textTracks, function(currentTrack) {
+        if (
+          (expectingStreamTextTracks || currentTrack.kind === 'captions') &&
+          // Manually added tracks have already been (or will be) added and notified
+          // when setClosedCaptions() is called, avoid mixing them up with in-manifest/in-stream tracks
+          !isExternalTextTrackId(currentTrack.id || currentTrack.trackId)
+        ) {
+          // For in-manifest/in-stream captions we use the id of the track (e.g.
+          // CC1, CC2, etc.) as a language in order to avoid conflicts with
+          // external captions.
+          var trackId = trySetStreamTextTrackId(currentTrack);
+          var captionInfo = {
+            language: trackId,
+            inStream: true,
+            label: currentTrack.label || currentTrack.language || 'Captions (' + trackId + ')'
```

Comment:
Where do we get 'Captions (' + trackId + ')' from?

Reply:
When there is no metadata we fallback to the `Captions (CC1)` format in order to mimic the behavior that we have on BitWrapper.

Reply:
Can you put the string in a temp var and then reference that inside this object declaration? (Code styleguide)

Reply:
Fixed